### PR TITLE
Added CSS to all forms #11

### PR DESCRIPTION
**Pull request by abdullah956**
1) Added JS in the base.html for the proper working of messages.
2) Added CSS to all the forms in the templates and messages.

### CODE REVIEW

File: templates/base.html [hunk 1]
```diff
@@ -4,6 +4,8 @@
 <head>
     <title>{% block title %}User Management{% endblock %}</title>
     <link rel="stylesheet" href="{% static 'css/bootstrap.min.css' %}">
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p" crossorigin="anonymous"></script>
```
Comment (muzzammilh) [inline templates/base.html]: What is the purpose of this file?

File: templates/dashboard/password_update.html [hunk 2]
```diff
@@ -1,19 +1,62 @@
...
+  {% if messages %}
+    <div class="messages">
+      {% for message in messages %}
+        {% if 'password_changed' in message.tags %}
+          <div class="alert alert-dismissible alert-success">
+            <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+            <strong>Success:</strong> {{ message }}
+          </div>
+        {% else %}
+          <div class="alert alert-dismissible {% if message.tags %}
```
Comment (muzzammilh) [inline templates/dashboard/password_update.html]: Don't break the tags like this, follow the example PR I created.

File: templates/users/login.html [hunk 3]
```diff
@@ -1,12 +1,30 @@
 {% extends 'base.html' %}
```
Comment (muzzammilh) [inline templates/users/login.html]: Login form is not working for invalid login. It doesn't show any kind of error messages.

**Review by muzzammilh — CHANGES_REQUESTED** [review]
Login form is not working for invalid information. HTML tags are line braked which needed to be fixed.

File: templates/base.html (hunk 1 above)
Comment (abdullah956) [reply templates/base.html]: > What is the purpose of this file?

To load this content on each template. the cross on the messages that will make it disappear on clicking was not working without js that why i added it ,we can remove the cross on the messages if we want to remove js.

**Comment by abdullah956** [comment]
> Login form is not working for invalid information. HTML tags are line braked which needed to be fixed.

fixed

File: templates/base.html [hunk 4]
```diff
@@ -4,6 +4,8 @@
 <head>
     <title>{% block title %}User Management{% endblock %}</title>
     <link rel="stylesheet" href="{% static 'css/bootstrap.min.css' %}">
+
```
Comment (muzzammilh) [inline templates/base.html]: Why we are adding extra blank spaces here?

File: templates/dashboard/password_update.html [hunk 5]
```diff
@@ -1,19 +1,53 @@
 {% extends 'base.html' %}
 
-{% block title %}Change Password{% endblock %}
+{% block title %}
```
Comment (muzzammilh) [inline templates/dashboard/password_update.html]: Why the original format is changed to be multiline?

File: templates/dashboard/password_update.html [hunk 6]
```diff
@@ -1,19 +1,53 @@
...
+  Change Password
+{% endblock %}
 
 {% block content %}
-<h2>Change Password</h2>
-{% if messages %}
-    <ul class="messages">
+  <h2>Change Password</h2>
+  {% if messages %}
```
Comment (muzzammilh) [inline templates/dashboard/password_update.html:7]: This block could be simple without nesting it. Taking advantage of Jinja template options like default. 

```
{% if messages %}
    <div class="messages">
        {% for message in messages %}
            <div class="alert alert-dismissible alert-{{ message.tags|default:"info" }}">
                <strong>{{ message.tags|capfirst }}:</strong> {{ message }}
            </div>
        {% endfor %}
    </div>
  {% endif %}
```

File: templates/users/login.html [hunk 7]
```diff
@@ -1,12 +1,37 @@
...
+          <div class="invalid-feedback">{{ error }}</div>
+        {% endfor %}
+      </div>
+      <div>
+        <label for="id_password" class="form-label mt-4">Password</label>
+        <input type="password" name="password" class="form-control {% if form.password.errors %}is-invalid{% endif %}" id="id_password" placeholder="Password" autocomplete="off" />
+        {% for error in form.password.errors %}
+          <div class="invalid-feedback">{{ error }}</div>
+        {% endfor %}
+        {% if form.non_field_errors %}
```
Comment (muzzammilh) [inline templates/users/login.html]: This is not the correct place to show generic non field error, ideally they need to be on the top. Above the email address container div.

**Review by muzzammilh — CHANGES_REQUESTED** [review]
Improvements suggested.

**Review by muzzammilh — APPROVED** [review]

**Pull request merged**